examples/upload_params.py

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env python3
+
+# Description: This is an example made by Ascend Engineering to provide parameters uploading capability by MAVSDK-Python and using .params file. 
+# Janitor in place: Farhang Naderi :)
+
+import asyncio,time,argparse
+from mavsdk import System
+#param_file_name = "file.params"
+parser = argparse.ArgumentParser()
+parser.add_argument("param_file", help="Param file to be uploaded with .params format")
+parser.add_argument("connection_type", help="The connection type, either 'udp' or 'serial' python3 upload_params_paramsync.py serial /dev/tty.usbmodem01:115200 or python3 upload_params_paramsync.py udp :14540")
+parser.add_argument("port", help="The port for the connection")
+
+args = parser.parse_args()
+system_address = f"{args.connection_type}://{args.port}"
+param_file_name = f"{args.param_file}"
+
+# Open the file and read it into a list of rows
+
+with open(param_file_name) as file:
+    for line in file:
+        if not line.startswith("#"):
+            columns = line.strip().split("\t")
+            vehicle_id = columns[0]
+            component_id = columns[1]
+            name = columns[2]
+            value = columns[3]
+            type = columns[4]
+
+
+async def set_params():
+    # Create a mavsdk.System object to access the system.
+    drone = System()
+
+    # Connect to the drone.
+    # Just leaving these here for later use in case args were not useful
+    #await drone.connect(system_address="udp://:14540")
+    #await drone.connect(system_address="serial:///dev/tty.usbmodem01:115200")
+    #await drone.connect(system_address="serial:///dev/ttyS0:921600")    # This is for a companion comp tried with RPi
+    await drone.connect(system_address=system_address)
+    print("Connected to the Vehicle")
+    # Create a mavsdk.Param object to access the parameters.
+    param_plugin = drone.param
+    # Get all of the parameters on the drone cached
+    params = await param_plugin.get_all_params()
+    float_params = params.float_params
+    int_params = params.int_params
+    custom_params = params.custom_params
+    int_param_names = [p.name for p in int_params]
+    float_param_names = [p.name for p in float_params]
+    custom_param_names = [p.name for p in custom_params]
+
+#Check if the vehicle is not flying and wait till it lands:
+    while True:
+        is_in_air = await drone.telemetry.in_air().__anext__()
```

Comment:
Why not using the async generator? I feel like it's not really supposed to be used with a `while True` and `__anext__()`.

```python
async for is_in_air in drone.telemetry.in_air():
    if is_in_air:
```

Reply:
thanks Jonas. Updated. I guess a mistake by me had happened so I chose the first solution. It is ok now.